### ISSUE TITLE
[feature request] [0.13] output all by default 

### ISSUE THREAD

**Issue by FernandoMiguel** [hashicorp/terraform#25240]
### Use-cases
I would love to see 0.13 output automatically for all resources, without the need to have the user explicitly create an output.

This would make modules outputs even easier.

### Proposal
automatically create outputs for all resources.
If the user creates an output with the same name, it overrides the implicit one.

by default do not print all the implicit output on apply, only explicit ones.
add cli-flag/env-var/terraform.rc option to print all (ex `--output-all`) 

this would allow things like CIs / atlantis where having too much out noise not a good thing, but when developing the dev can print all outputs and debug easier.
would also make all modules super easier to see their outputs used in the root module.

**Comment by FernandoMiguel** [comment]
A bit of historical context 
https://github.com/hashicorp/terraform/issues/16397#issuecomment-402782209

> blindly exporting large objects up to the root module is likely to cause more coupling between modules that could be harder to maintain in the long run

As we move to 0.13, and treat modules better, there no reason to leave them as second class citizens, while any other resource in the same level module can access another resource attributes

**Comment by danieldreier** [comment]
@FernandoMiguel there are two questions here:

1. Release vehicle for this (you requested this for 0.13)
2. The feature request itself

The release vehicle for this will definitely not be 0.13.0, because 0.13 beta 2 (released today)is when we hit our 0.13.0 feature freeze, and I think (not 100%, but think) that adding default output all would constitute a breaking change because it would be a significant behavior change. From here on out, we are only fixing bugs and adding no new functionality to 0.13.0.

I'm chiming in because you messaged me in hangops slack, but I actually don't know the design history of this. My first response is that explicitly declaring relationships between modules makes sense to me, because at least some terraform modules have resources that are logically considered internal and others are public interfaces. The use case I'm imagining here is organizations where a central function (cloud team, ops team, etc) provides a module to app services teams and wants to be able to refactor it and know which resources people might be making references to and which not. 

That said, that's my personal hot take and not necessarily the stance that Terraform needs to take forever. @pkolyvas and @apparentlymart are better people to chime in on whether there are other requests for this and what the rationale is for not exposing everything already. @apparentlymart I think that since this was requested in https://github.com/hashicorp/terraform/issues/16397 and the current functionality was basically an experiment to see if it's sufficient, this is probably worth at least reconsidering to see whether it should go into 0.14 planning.

**Comment by FernandoMiguel** [comment]
@danieldreier so 0.14? :D

**Comment by danieldreier** [comment]
No, this isn't in 0.14, and also isn't looking likely to be prioritized for 0.15; we're trying to pull forward the minimal set of breaking changes needed to get to 1.0, so that we can have stable 1.x releases, and so a lot of design work is on hold in favor of avoiding backward-incompatible changes during the 1.x lifecycle.
